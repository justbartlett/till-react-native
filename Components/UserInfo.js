import React from 'react';
import {Colors} from '../Theme/Colors';
import {DATA} from '../Services/Transactions';
import {StyleSheet, View, Text, Image} from 'react-native';

const UserInfo = () => {
  return (
    <View style={styles.userInfoWrapper}>
      <View style={styles.userInfo}>
        <View>
          <Image
            source={{
              uri: DATA.avatar,
            }}
            style={styles.avatar}
          />
        </View>
        <View>
          <Text style={styles.name}>{DATA.name}</Text>
          <Text style={styles.balance}>My Balance: ${DATA.balance}</Text>
        </View>
      </View>
    </View>
  );
};

export default UserInfo;

const styles = StyleSheet.create({
  userInfo: {
    flex: 1,
    flexDirection: 'row',
  },
  userInfoWrapper: {height: 80, marginLeft: '10%'},
  avatar: {
    marginRight: 15,
    width: 52,
    height: 52,
    borderRadius: 100,
    borderWidth: 4,
    borderColor: Colors.white,
  },
  name: {
    color: Colors.white,
    fontSize: 22,
    marginBottom: 4,
    fontFamily: 'NunitoSans-Black',
  },
  balance: {
    color: Colors.white,
    fontSize: 13,
    fontFamily: 'NunitoSans-ExtraBold',
  },
});
