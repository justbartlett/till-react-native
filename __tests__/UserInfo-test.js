/**
 * @format
 */

import 'react-native';
import React from 'react';
import UserInfo from '../Components/UserInfo';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<UserInfo />).toJSON();
  expect(tree).toMatchSnapshot();
});
