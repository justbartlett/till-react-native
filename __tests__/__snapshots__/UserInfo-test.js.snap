// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "height": 80,
      "marginLeft": "10%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View>
      <Image
        source={
          Object {
            "uri": "http://www.fillmurray.com/200/200",
          }
        }
        style={
          Object {
            "borderColor": "#FFFFFF",
            "borderRadius": 100,
            "borderWidth": 4,
            "height": 52,
            "marginRight": 15,
            "width": 52,
          }
        }
      />
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "NunitoSans-Black",
            "fontSize": 22,
            "marginBottom": 4,
          }
        }
      >
        Johnny Ray
      </Text>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "NunitoSans-ExtraBold",
            "fontSize": 13,
          }
        }
      >
        My Balance: $
        998913.61
      </Text>
    </View>
  </View>
</View>
`;
